service/src/main/java/greencity/service/EventServiceImpl.java

```diff
@@ -0,0 +1,216 @@
+    private final UserService userService;
+    private final RestClient restClient;
+    private final FileService fileService;
+    ModelMapper modelMapper = new ModelMapper();
+
+    @Override
+    public EventCreateDtoResponse create(EventCreateDtoRequest dto, MultipartFile[] images, String principal) {
+
+        List<Event> fetchedEvents = eventRepo.findByTitle(dto.getTitle());
+        checkIfEventExistsOrElseThrow(dto, fetchedEvents);
+
+        var organiser = userService.findByEmail(principal);
+
+        /**
+         * If no @images: titleImage set to empty String, additionalImages set to empty List
+         * If 1 @images:  this image is set to titleImage, additionalImages set to empty List
+         * If >1 @images: first image is set to titleImage, others are set to additionalImages
+         */
+        var additionalImagesLinks = getLinksOfImages(images);
+        var titleImage = additionalImagesLinks.isEmpty() ? "" : additionalImagesLinks.getFirst().getData();
+        var additionalImageList = additionalImagesLinks.size() > 1 ?
+                additionalImagesLinks.subList(1, additionalImagesLinks.size()) : new ArrayList<AdditionalImage>();
+
+        Event event = Event.builder()
+                .title(dto.getTitle())
+                .description(dto.getDescription())
+                .titleImage(titleImage)
+                .isOpen(dto.getIsOpen())
+                .organizer(User.builder()
+                        .id(organiser.getId())
+                        .name(organiser.getName())
+                        .rating(organiser.getRating())
+                        .email(organiser.getEmail())
+                        .build())
+                .build();
+        for (AdditionalImage a : additionalImageList) {
+            a.setEvent(event);
+        }
+        event.setAdditionalImages(additionalImageList);
+
+        List<EventDateLocation> eventDateLocationList = new ArrayList<>();
+        for (EventDateLocationDtoRequest e : dto.getDates()) {
+            EventDateLocation eventDateLocation = modelMapper.map(e, EventDateLocation.class);
+            Address address = modelMapper.map(e.getAddress(), Address.class);
+
+            eventDateLocation.setAddress(address);
+            eventDateLocation.setEvent(event);
+
+            eventDateLocationList.add(eventDateLocation);
+        }
+        event.setDates(eventDateLocationList);
+        event = eventRepo.save(event);
+
+        restClient.sendNotificationToUser(prepareNotificationFromEvent(event), organiser.getEmail());
+
+        return modelMapper.map(event, EventCreateDtoResponse.class);
+    }
+
+    /**
+     * Deletes an event with the specified eventId and name.
+     *
+     * @param eventId The ID of the event to be deleted.
+     * @param email The email of the user who is deleting the event.
+     */
+    @Override
+    public void delete(Long eventId, String email) {
+        UserVO userVO = restClient.findByEmail(email);
```

Comment:
You can use userService.findByEmail() instead sending http request to User Service